src/services/importService.ts

```diff
@@ -0,0 +1,186 @@
+import shortid from "shortid";
+import { IProject, IAppSettings, ITag, IConnection,
+         IV1Project, IAssetMetadata, IRegion, RegionType, AssetType, AssetState } from "../models/applicationState";
+import { AssetService } from "./assetService";
+import MD5 from "md5.js";
+import { randomIntInRange } from "../common/utils";
+import TagColors from "../react/components/common/tagsInput/tagColors.json";
+// const TagColors = require("../react/components/common/tagsInput/tagColors.json");
+
+/**
+ * Functions required for an import service
+ * @member convertV1 - Converts a v1 project to v2 project
+ * @member generateAssets - Generates v2 assets based on v1 project file
+ * @member generateConnections - Generates v2 connections based on location of v1 project file
+ */
+export interface IImportService {
+    convertV1(project: any): Promise<IProject>;
+}
+
+/**
+ * @name - Import Service
+ * @description - Functions for importing v1 projects to v2 application
+ */
+// change any to { content, file } object eventually
+export default class ImportService implements IImportService {
+    public convertV1(project: any): Promise<IProject> {
+        return new Promise<IProject>((resolve, reject) => {
```

Comment:
You should be able to use standard `async/await` here vs having to wrap this in a promise.

Reply:
Doesn't any async method have to return a promise?

Reply:
Yes, async methods return a promise... but if your method is making other async calls using await then your return value will already be wrapped in a promise.